src/hooks/useApproveCallback.ts

```diff
@@ -9,53 +9,63 @@ import { computeSlippageAdjustedAmounts } from '../utils/prices'
 import { calculateGasMargin } from '../utils'
 import { useTokenContract, useWeb3React } from './index'
 
-// returns a function to approve the amount required to execute a trade if necessary, otherwise null
-export function useApproveCallback(trade?: Trade, allowedSlippage = 0): [undefined | boolean, () => Promise<void>] {
+// returns a boolean indicating whether approval is necessary, and a function to approve if it is
```

Comment:
what does undefined mean for the boolean? can we be more explicit `false | [boolean, fn]`

Reply:
`undefined` means the current approval amount is yet to be fetched. `false | [boolean, fn]` wouldn't play nicely with destructuring. if we want to be more explicit we could make the first arg an enum?

Reply:
yeah that's better, but i'm also ok with what it is now too

Reply:
for the record, made it an enum and as a bonus we now get pending state from this hook too